controller/node_controller.go

```diff
@@ -561,6 +561,8 @@ func (nc *NodeController) enqueueKubernetesNode(obj interface{}) {
 }
 
 func (nc *NodeController) syncDiskStatus(node *longhorn.Node, collectedDataInfo map[string]*monitor.CollectedDiskInfo) error {
+	cleanupDiskStatusOldStorageData(node)
```

Comment:
Why not directly call `alignDiskSpecAndStatus` here instead?

Reply:
Yes, we can do this.

I'm thinking whether we should return error directly if `nc.syncWithMonitor` failed no matter what reason. But this might introduce many retries while reconciling the node CR.
https://github.com/longhorn/longhorn-manager/blob/master/controller/node_controller.go#L437

Currently, we didn't return error and just align the disk spec and status if `nc.syncWithMonitor` failed. Thus, I prefer to align the disk spec and status first before exiting the reconciliation.

Reply:
- Yes. It's better not to return errors for the mismatching case.
- The reason I prefer to put (previous) `alignDiskSpecAndStatus` here is that the node controller can handle all disk related operations inside `syncDiskStatus`.

Reply:
It seems no hurt and one place to handle that. The same suggestion, move alighDiskSpecAndStatus to the syncDiskStatus. 

But this is a good finding to avoid unintentionally updating the wrong disk status.

Reply:
> The reason I prefer to put (previous) alignDiskSpecAndStatus here is that the node controller can handle all disk related operations inside syncDiskStatus.

Sure. I agree your opinion.

BTW, looks we cannot know if diskUpdate (in test_node.py) takes effect or not… and there is no retry mechanism in diskUpdate. I have no better idea for the retry.
The mismatching disk spec and status because of the time delay leads to the false alarm in test_node.py